kernel/fs/fs_socket.c

```diff
@@ -586,6 +586,25 @@ int getsockname(int sock, struct sockaddr *name, socklen_t *name_len) {
     return hnd->protocol->getsockname(hnd, name, name_len);
 }
 
+int getpeername(int sock, struct sockaddr *__RESTRICT name, socklen_t *__RESTRICT name_len) {
+    net_socket_t *hnd;
+
+    hnd = (net_socket_t *)fs_get_handle(sock);
+
+    if(hnd == NULL) {
+        errno = EBADF;
+        return -1;
+    }
+
+    /* Make sure this is actually a socket. */
+    if(fs_get_handler(sock) != &vh) {
+        errno = ENOTSOCK;
+        return -1;
+    }
+
+    return hnd->protocol->getpeername(hnd, name, name_len);
```

Comment:
I know it was implemented for the two protocols we have, but this should check if protocol has `getpeername` and if not throw `EOPNOTSUPP`